### PR TITLE
[server] A/A + WC should handle update on key that were written in the batch phase

### PR DESCRIPTION
**Pull request by sixpluszero**
## [server] A/A + WC should handle update on key that were written in the batch phase

This PR fixes a bug that MergeConflictResolver.update() will throw exception when it sees oldValue without a RMD.
The assumption is wrong because in the batch phase, AASIT call LFSIT logic to process batch data without creating RMD value. The fix is to find the correct writer schema for the original value and create the new field-level RMD schema.

## How was this PR tested?

Add new integration test.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by sixpluszero** [comment]
This PR fixes a bug that MergeConflictResolver.update() will throw exception when it sees oldValue without a RMD.
The assumption is wrong because in the batch phase, AASIT call LFSIT logic to process batch data without creating RMD value. The fix is to find the correct writer schema for the original value and create the new field-level RMD schema.

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java [hunk 1]
```diff
@@ -140,6 +141,97 @@ public void setUp() {
     this.parentController = parentControllers.get(0);
   }
 
+  @Test
+  public void testTest() {
```
Comment (adamxchen) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:145]: Can we have a better name on this?

Comment (sixpluszero) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java:145]: Oh right this test should not be part of the PR, I will remove it.

**Review by majisourav99 — APPROVED** [review]
LGTM

**Pull request merged**